src/zope/app/testing/dochttp.py

```diff
@@ -79,31 +80,34 @@ def dochttp(args=sys.argv[1:], default=None):
     names = [name[:-len(".request")]
              for name in os.listdir(directory)
              if name.startswith(options.prefix) and name.endswith('.request')
-             ]
+    ]
     names.sort()
 
-    extre = re.compile("[.](\w+)$")
+    extre = re.compile(r"[.](\w+)$")
 
     for name in names:
-        requests =  Requests(
-                        open(os.path.join(directory, name + ".request"), 'rb'),
-                        options.skip_request_header,
-                        )
-        responses = Responses(
-                        open(os.path.join(directory, name + ".response"), 'rb'),
-                        options.skip_response_header,
-                        )
-
-        # We use map so as *not* to truncate at shortest input.
-        # We want an error if the number of requests and responses
-        # is different.
-        for request, response in map(None, requests, responses):
+        with open(os.path.join(directory, name + ".request"), 'rb') as f:
+            requests = list(Requests(
+                f,
+                options.skip_request_header,
+            ))
+        with open(os.path.join(directory, name + ".response"), 'rb') as f:
+            responses = list(Responses(
+                f,
+                options.skip_response_header,
+            ))
+
+        if len(requests) != len(responses):
+            raise ValueError("Expected equal length requests and responses")
+
+        for request, response in zip(requests, responses):
             assert (request and response) or not (request or response)
```

Comment:
I think this assert shouldn't be needed any more, now that you verify that `len(requests) == len(responses)`.

(I'm somewhat baffled why it would allow both request and response to be `None`.)

Reply:
The request and response objects define a custom `__bool__` method to determine how many HTTP messages there are, if any, so that's still a useful check.

Reply:
Ah, ok!